learn/chapter1/4-shaders.md

```diff
@@ -0,0 +1,272 @@
+# Shaders
+
+As mentioned in the Hello Triangle [Hello Triangle](2-hello-triangle.md) chapter, shaders are little programs that rest on the GPU. These programs are run for each specific section of the graphics pipeline. In a basic sense, shaders are nothing more than programs transforming inputs to outputs. Shaders are also very isolated programs in that they're not allowed to communicate with each other; the only communication they have is via their inputs and outputs.
+
+In the previous chapter we briefly touched the surface of shaders and how to properly use them. We will now explain shaders, and specifically the OpenGL Shading Language, in a more general fashion. 
+
+# GLSL
+
+ Shaders are written in the C-like language GLSL. GLSL is tailored for use with graphics and contains useful features specifically targeted at vector and matrix manipulation.
+
+Shaders always begin with a version declaration, followed by a list of input and output variables, uniforms and its main function. Each shader's entry point is at its `main` function where we process any input variables and output the results in its output variables. Don't worry if you don't know what uniforms are, we'll get to those shortly.
+
+A shader typically has the following structure: 
+
+```glsl
```

Comment:
Some markdown doesn't recognise `glsl`, you might have to change this to `c`, but you might not know that until it's rendered.

Reply:
I saw glsl mark in other sections of the documentation and had a look on the main webpage and it seemed to look fine there

Reply:
:+1: